pam/internal/pam_test/service-utils_test.go

```diff
@@ -0,0 +1,86 @@
+package pam_test
+
+import (
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestCreateService(t *testing.T) {
+	t.Parallel()
+
+	tests := map[string]struct {
+		services    []ServiceLine
+		wantContent string
+	}{
+		"empty":         {},
+		"CApital-Empty": {},
+		"auth-sufficient-permit": {
+			services: []ServiceLine{
+				{Auth, Sufficient, Permit.String(), []string{}},
+			},
+			wantContent: "auth	sufficient	pam_permit.so",
+		},
+		"auth-sufficient-permit-args": {
+			services: []ServiceLine{
+				{Auth, Required, Deny.String(), []string{"a b c [d e]"}},
+			},
+			wantContent: "auth	required	pam_deny.so	[a b c [d e\\]]",
+		},
+		"account-sufficient-requisite": {
+			services: []ServiceLine{
+				{Auth, SufficientRequisite, Permit.String(), []string{}},
+			},
+			wantContent: "auth	[success=done new_authtok_reqd=done ignore=ignore default=die]	pam_permit.so",
+		},
+		"complete-custom": {
+			services: []ServiceLine{
+				{Account, Required, "pam_account_module.so", []string{"a", "b", "c", "d e", "f [g h]"}},
+				{Account, Required, Deny.String(), []string{}},
+				{Auth, Requisite, "pam_auth_module.so", []string{}},
+				{Auth, Requisite, Deny.String(), []string{}},
+				{Password, Sufficient, "pam_password_module.so", []string{"arg"}},
+				{Password, Sufficient, Deny.String(), []string{}},
+				{Session, Optional, "pam_session_module.so", []string{""}},
+				{Session, Optional, Ignore.String(), []string{}},
+			},
+			wantContent: `account	required	pam_account_module.so	a b c [d e] [f [g h\]]
+account	required	pam_deny.so
+auth	requisite	pam_auth_module.so
+auth	requisite	pam_deny.so
+password	sufficient	pam_password_module.so	arg
+password	sufficient	pam_deny.so
+session	optional	pam_session_module.so
+session	optional	pam_debug.so auth=incomplete cred=incomplete acct=incomplete prechauthtok=incomplete chauthtok=incomplete open_session=incomplete close_session=incomplete`,
+		},
+	}
+
+	for name, tc := range tests {
+		tc := tc
+		name := name
+		t.Run(name, func(t *testing.T) {
+			t.Parallel()
+			service, err := CreateService(t.TempDir(), name, tc.services)
+			require.NoError(t, err, "Can't create service file")
+
+			require.Equal(t, strings.ToLower(name), filepath.Base(service),
+				"Invalid service name %s", service)
```

Comment:
As the `require.Equal()` is still related to the service creation, the blocks should be:
```Go
            t.Parallel()

			service, err := CreateService(t.TempDir(), name, tc.services)
			require.NoError(t, err, "Can't create service file")
			require.Equal(t, strings.ToLower(name), filepath.Base(service),
				"Invalid service name %s", service)
```